milestones/serializers.py

```diff
@@ -0,0 +1,67 @@
+"""
+Data layer serialization operations.  Converts querysets to simple
+python containers (mainly arrays and dicts).
+"""
+from . import models
+
+
+def serialize_milestone(milestone):
+    """
+    Milestone object-to-dict serialization
+    """
+    return {
+        'id': milestone.id,
+        'name': milestone.name,
+        'namespace': milestone.namespace,
+        'description': milestone.description
+    }
+
+
+def serialize_milestone_with_course(course_milestone):
+    """
+    CourseMilestone serialization (composite object)
+    """
+    return {
+        'id': course_milestone.milestone.id,
+        'name': course_milestone.milestone.name,
+        'namespace': course_milestone.milestone.namespace,
+        'description': course_milestone.milestone.description,
+        'course_id': course_milestone.course_id
+    }
```

Comment:
My main concern is for us to know when we're going to hit performance issues. Since the serializer will dereference 'milestones', it'd be good for us to know if that is going to cause a DB fetch, particularly if this is being called in a loop over a collection.

More of a discussion/curiosity than any particular feedback.